app/frontend/components/pages/delegations/delegateInput.tsx

```diff
@@ -29,86 +24,18 @@ const StakePoolLabel = ({
   </Fragment>
 )
 
-// TODO: move to types
-type Error = {
-  code: string
-  params?: {hasTickerMapping: boolean}
+type Props = {
+  value: string
+  onChange: (event: any) => any
 }
 
-type ValidatedInput = {
-  poolHash: string | null
-  error: Error
-}
-
-const validateInput = (
-  fieldValue: string,
-  validStakepoolDataProvider: StakepoolDataProvider
-): ValidatedInput => {
-  if (ADALITE_CONFIG.ADALITE_ENABLE_SEARCH_BY_TICKER) {
-    const pool =
-      validStakepoolDataProvider.getPoolInfoByPoolHash(fieldValue) ||
-      validStakepoolDataProvider.getPoolInfoByTicker(fieldValue)
-    if (pool) return {poolHash: pool.poolHash, error: null}
-
-    const hasTickerMapping = validStakepoolDataProvider.hasTickerMapping
-    const isTickerString = fieldValue.length <= 5 && fieldValue.toUpperCase() === fieldValue
-    const poolHash = null
-    if (!hasTickerMapping && isTickerString) {return {poolHash, error: {code: 'TickerSearchDisabled'}}}
-    return {poolHash, error: {code: 'InvalidStakepoolIdentifier', params: {hasTickerMapping}}}
-  }
-
-  const pool = validStakepoolDataProvider.getPoolInfoByPoolHash(fieldValue)
-  if (pool) return {poolHash: pool.poolHash, error: null}
-  return {
-    poolHash: null,
-    error: {code: 'InvalidStakepoolIdentifier', params: {hasTickerMapping: false}},
-  }
-}
-
-interface Props {
-  poolRecommendation: PoolRecommendation
-  pool: Stakepool
-  validStakepoolDataProvider: StakepoolDataProvider
-  updateStakePoolIdentifier: (poolHash: string, validationError?: any) => void
-  resetStakePoolIndentifier: () => void
-}
-
-const DelegateInput = ({
-  poolRecommendation,
-  pool,
-  validStakepoolDataProvider,
-  updateStakePoolIdentifier,
-  resetStakePoolIndentifier,
-}: Props): h.JSX.Element => {
-  const [fieldValue, setFieldValue] = useState('')
```

Comment:
General recommendation. Do not get into flow like:
"child" -> "has local state and sync it with global" -> "parent" -> "read from global"

...just move the state to parent and avoid global state if not really necessary.

Reply:
In general "parent" (usualy the form screen) should be responsible for storing form state, not the form fields itself.